### PR TITLE
Add new legend property for time and list charts

### PR DESCRIPTION
**Pull request by cory-signalfx**
# Summary

Add a new property for controlling both order and presence of data table legend fields.

# Motivation

As discussed in #13 there is no way to hide/show fields and order them. This is because, by default, the chart shows *all* dimensions and the existing `legend_fields_to_hide` is a convenience property to hide fields.

# Notes

* ~I need to verify that order is supported, as it isn't in the UI.~
* No tests, yet.

### CODE REVIEW

**Comment by tiedotguy** [comment]
Order is supported in the UI, I can drag and drop to re-order the list under Data Table Columns, which also also flows through to the tooltip.

File: signalfx/util.go [hunk 1]
```diff
@@ -276,6 +276,34 @@ func getLegendOptions(d *schema.ResourceData) map[string]interface{} {
...
+		for i, field := range fields {
+			property := field.Get("property_name").(string)
+			if property == "metric" {
+				property = "sf_originatingMetric"
+			} else if property == "plot_label" || property == "Plot Label" {
+				property = "sf_metric"
+			}
+			item := make(map[string]interface{})
+			item["property"] = property
+			item["enabled"] = false
```
Comment (tiedotguy) [inline signalfx/util.go]: Not sure if this is meant as a WIP or ready for review, but missing the all important path for true.

Comment (cory-signalfx) [reply signalfx/util.go]: Very much a WIP, but thank you for flagging. Can never be too safe. ;)

**Comment by cory-signalfx** [comment]
Ok, this should be good to go!

**Pull request merged**